pkg/roachprod/failureinjection/failures/failure.go

```diff
@@ -107,3 +121,134 @@ func (f *GenericFailure) NetworkInterfaces(
 	}
 	return f.networkInterfaces, nil
 }
+
+func getDiskDevice(ctx context.Context, f *GenericFailure, l *logger.Logger) error {
+	if f.diskDevice.name == "" {
+		res, err := f.c.RunWithDetails(ctx, l, install.WithNodes(f.c.Nodes[:1]), "Get Disk Device", "lsblk -o NAME,MAJ:MIN,MOUNTPOINTS | grep /mnt/data1 | awk '{print $1, $2}'")
+		if err != nil {
+			return errors.Wrapf(err, "error when determining block device")
+		}
+		parts := strings.Split(strings.TrimSpace(res[0].Stdout), " ")
+		if len(parts) != 2 {
+			return errors.Newf("unexpected output from lsblk: %s", res[0].Stdout)
+		}
+		f.diskDevice.name = strings.TrimSpace(parts[0])
+		major, minor, found := strings.Cut(parts[1], ":")
+		if !found {
+			return errors.Newf("unexpected output from lsblk: %s", res[0].Stdout)
+		}
+		if f.diskDevice.major, err = strconv.Atoi(major); err != nil {
+			return err
+		}
+		if f.diskDevice.minor, err = strconv.Atoi(minor); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (f *GenericFailure) DiskDeviceName(ctx context.Context, l *logger.Logger) (string, error) {
+	if err := getDiskDevice(ctx, f, l); err != nil {
+		return "", err
+	}
+	return "/dev/" + f.diskDevice.name, nil
+}
+
+func (f *GenericFailure) DiskDeviceMajorMinor(
+	ctx context.Context, l *logger.Logger,
+) (int, int, error) {
+	if err := getDiskDevice(ctx, f, l); err != nil {
+		return 0, 0, err
+	}
+	return f.diskDevice.major, f.diskDevice.minor, nil
+}
+
+func (f *GenericFailure) PingNode(
+	ctx context.Context, l *logger.Logger, nodes install.Nodes,
+) error {
+	// TODO(darryl): Consider having failure modes accept a db connection pool
+	// in makeFailureFunc() or having each failureMode manage it's own pool.
+	res, err := f.c.ExecSQL(
+		ctx, l, nodes, install.SystemInterfaceName,
+		0, install.AuthUserCert, "", /* database */
+		[]string{"-e", "SELECT 1"},
+	)
+
+	return errors.CombineErrors(err, res[0].Err)
+}
+
+func (f *GenericFailure) WaitForSQLReady(
+	ctx context.Context, l *logger.Logger, node install.Nodes, timeout time.Duration,
+) error {
+	start := timeutil.Now()
+	err := retryForDuration(ctx, timeout, func() error {
+		if err := f.PingNode(ctx, l, node); err == nil {
+			l.Printf("Connected to node %d after %s", node, timeutil.Since(start))
+			return nil
+		}
+		return errors.Newf("unable to connect to node %d", node)
+	})
+
+	return errors.Wrapf(err, "never connected to node %d after %s", node, timeout)
+}
+
+func (f *GenericFailure) WaitForSQLNodeDeath(
```

Comment:
Nit: `WaitForSQLUnavailable`. "Death" implies the process is down, which `PingNode` can't establish; so, the current name is somewhat misleading.